types/mongodb/index.d.ts

```diff
@@ -858,6 +858,17 @@ export interface FSyncOptions extends CommonOptions {
 
 type OptionalId<TSchema> = Omit<TSchema, '_id'> & { _id?: any };
 
+type ExtractIdType<TSchema> =
+    TSchema extends { _id: infer U } // user has defined a type for _id
+    ? ({} extends U ? Exclude<U, {}> : U) // Exclude is used here to fix typescript 2 bug when TSchema is "any"
+    // TODO: uncomment the following and remove the above line after updating to TypeScript v3.0
+    // ? (unknown extends U ? ObjectId : U)
```

Comment:
Instead of putting TODOs in the code, can you please either:

- Update the minimum TypeScript version in the header and use the version with `unknown`
- Remove the commented out version and file an issue

Reply:
Thanks @mrcrane for the review. I completely agree. I created this https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38686 and this https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38687 and removed the comment.